### PR TITLE
Switch to using Visual Studio's version of LLVM and Clang

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

This pull request includes changes to the `.github/workflows/reusable-build.yml` file to streamline the build process by removing unnecessary steps and adding new ones for setting up Visual Studio LLVM.

Build process improvements:

* Removed setup for the choco cache folder and caching of choco packages. (`.github/workflows/reusable-build.yml`)
* Added a step to add Visual Studio LLVM to the PATH. (`.github/workflows/reusable-build.yml`)
* Replaced the installation of LLVM tools with a check for the Clang version. (`.github/workflows/reusable-build.yml`)

## Testing

CI/CD

## Documentation

TBD

## Installation

No.

### CODE REVIEW

File: .github/workflows/reusable-build.yml
Comment (dthaler) [inline .github/workflows/reusable-build.yml:139]: I don't think we should hard code path names in the middle of job commands, use variables instead (yes i see line 160 already did previously so it's not new in this PR).

File: docs/GettingStarted.md
Comment (dthaler) [inline docs/GettingStarted.md:236]: Does LLVM 17 work with cilium?  I thought it didn't.

File: ebpf-for-windows.sln
Comment (dthaler) [inline ebpf-for-windows.sln:222]: Why did the GUID change?

File: docs/GettingStarted.md
Comment (Alan-Jowett) [reply docs/GettingStarted.md:236]: Unfortunately, supply chain security trumps this. Folks consuming the Cilium BPF programs we will either need to fix prevail or the BPF programs.

We cannot deploy a production version of our products based on a code base that isn't getting security fixes. No one is signed up to provide security fixes for LLVM 11.0.1, but Microsoft's Visual Studio team is providing security fixes for LLVM 17.0.3 (at least the version that ships with Visual Studio).

File: ebpf-for-windows.sln
Comment (Alan-Jowett) [reply ebpf-for-windows.sln]: Reverted.

File: .github/workflows/reusable-build.yml
Comment (Alan-Jowett) [reply .github/workflows/reusable-build.yml]: Switch to derive clang.exe path from cl.exe path.

File: docs/GettingStarted.md
Comment (dthaler) [reply docs/GettingStarted.md:236]: Sounds like a good reason to me.

File: scripts/Setup-DevEnv.ps1
Comment (dthaler) [inline scripts/Setup-DevEnv.ps1:11]: This should not assume that `%ProgramFiles%` is on the C drive.

Comment (Alan-Jowett) [reply scripts/Setup-DevEnv.ps1:11]: Good catch. Let me see if I can apply the same fix as I did to the yaml.

**Review by dthaler — DISMISSED** [review]

File: docs/GettingStarted.md
Comment (Alan-Jowett) [reply docs/GettingStarted.md:236]: With a few very minor bug fixes in the Cilium code, it now compiles using LLVM 17!

**Review by dthaler — APPROVED** [review]

**Review by shpalani — DISMISSED** [review]

**Comment by Alan-Jowett** [comment]
@shpalani and @dthaler please re-approve after I had to resolve a merge conflict from main.

**Review by shpalani — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**
